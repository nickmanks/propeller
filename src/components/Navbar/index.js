/* eslint complexity: 0 */

import React, {useState} from 'react';
import logo from '#src/logo.svg';
import logoWhite from '#src/logo.svg';
import styles from './index.scss';

const Navbar = ()=> {
  const [open, setOpen] = useState(false);

  return (
    <>
      <div
        className={
          open ? `${styles.overlay} ${styles.overlayOpen}` : styles.overlay
        }
      >
        <div className={styles.overlayContent}>
          <a>About</a>
          <a>Services</a>
          <a>Get Quote</a>
          <a>Contact</a>
        </div>
      </div>
      <div
        className={
          open
            ? `${styles.container} ${styles.containerOpen}`
            : styles.container
        }
      >
        <div className={styles.logo}>
          <img
            className={open ? styles.image : undefined}
            src={open ? logoWhite : logo}
            width="130"
          />
        </div>
        <nav className={open ? `${styles.nav} ${styles.navOpen}` : styles.nav}>
          <ul>
            <li>Login</li>
            <li>
              <button
                className={
                  open ? `${styles.button} ${styles.buttonOpen}` : styles.button
                }
              >
                Sign Up
              </button>
            </li>
            <li onClick={()=> setOpen(!open)}>
              <div className={open ? styles.one : undefined}></div>
              <div className={open ? styles.two : undefined}></div>
              <div className={open ? styles.three : undefined}></div>
            </li>
          </ul>
        </nav>
      </div>
    </>
  );
};

export default Navbar;
