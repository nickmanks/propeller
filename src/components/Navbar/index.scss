@import '#scss/colors';

.container {
  align-items: center;
  background: $primary;
  display: flex;
  height: 10vh;
  justify-content: space-between;
  margin: auto;
  padding-left: 5%;
  padding-right: 5%;
  position: relative;
  top: 0;
  transition: 0.2s;
  z-index: 101;

  &-open {
    background: $primary;
  }
}

.nav {
  ul {
    cursor: pointer;
    list-style-type: none;
    margin: 0;
    overflow: hidden;
    padding: 0;
    position: relative;
    z-index: 101;

    li {
      color: $secondary;
      float: left;
      font-size: 20px;
      margin-right: 2rem;
      margin-top: 1rem;
      transition: 0.2s;

      &:last-child {
        margin-right: 0;
      }

      div {
        background-color: $secondary;
        border-radius: 5rem;
        height: 3px;
        margin: 4px 0;
        transition: 0.2s;
        width: 24px;
      }
    }
  }

  &-open {
    ul {
      li {
        color: $secondary;

        div {
          background-color: $secondary;
        }
      }
    }
  }
}

.image {
  cursor: pointer;
  position: relative;
  transition: 0.3s;
  z-index: 101;
}

.button {
  background-color: #fff;
  border: unset;
  border-radius: 5rem;
  color: $secondary;
  cursor: pointer;
  font-size: 20px;
  margin-top: -1rem;
  padding: 0.5rem 2rem;

  &:hover {
    background-color: darken(#fff, 3);
    box-shadow: 0 3px 4px rgba(0, 0, 0, 0.1);
  }

  &-open {
    background-color: $secondary;
    color: #fff;

    &:hover {
      background-color: lighten($secondary, 2);
      box-shadow: 0 3px 4px rgba(0, 0, 0, 0.1);
    }
  }
}

.one {
  transform: rotate(-45deg) translate(-4px, 4px);
}

.two {
  opacity: 0;
}

.three {
  transform: rotate(45deg) translate(-6px, -6px);
}

.overlay {
  background-color: $primary;
  height: 0;
  left: 0;
  overflow-x: hidden;
  position: fixed;
  top: 0;
  transition: 0.2s;
  width: 100%;
  z-index: 100;

  &-open {
    height: 100%;
    max-height: 100vh;
  }
}

.overlay-content {
  border-bottom: 1px solid $secondary;
  margin: auto;
  margin-top: 30px;
  padding-bottom: 3rem;
  position: relative;
  text-align: center;
  top: 25%;
  width: 100%;
  width: 15rem;

  a {
    cursor: pointer;
  }
}

.overlay a {
  color: $secondary;
  display: inline-block;
  font-size: 36px;
  padding: 8px;
  text-decoration: none;
  transition: 0.2s;

  &::after {
    background: none repeat scroll 0 0 transparent;
    background: $secondary;
    content: '';
    display: block;
    height: 2px;
    left: 50%;
    position: absolute;
    transition: width 0.3s ease 0s, left 0.3s ease 0s;
    width: 0;
  }

  &:hover::after {
    left: 0;
    margin: auto;
    right: 0;
    width: 11rem;
  }
}
