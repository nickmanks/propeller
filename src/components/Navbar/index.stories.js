import React from 'react';
import Navbar from '.';
import {component} from '#src/testing/snapshots';

component('Navbar').when('standard', ()=> <Navbar />);
