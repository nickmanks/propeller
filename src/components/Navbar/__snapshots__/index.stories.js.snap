// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`standard 1`] = `
Array [
  <div
    className="mock-overlay"
  >
    <div
      className="mock-overlayContent"
    >
      <a>
        About
      </a>
      <a>
        Services
      </a>
      <a>
        Get Quote
      </a>
      <a>
        Contact
      </a>
    </div>
  </div>,
  <div
    className="mock-container"
  >
    <div
      className="mock-logo"
    >
      <img
        src="logo.png"
        width="130"
      />
    </div>
    <nav
      className="mock-nav"
    >
      <ul>
        <li>
          Login
        </li>
        <li>
          <button
            className="mock-button"
          >
            Sign Up
          </button>
        </li>
        <li
          onClick={[Function]}
        >
          <div />
          <div />
          <div />
        </li>
      </ul>
    </nav>
  </div>,
]
`;
